composer/trainer/trainer.py

```diff
@@ -582,20 +583,21 @@ def __init__(
             steps_per_epoch=train_subset_num_batches,
         )
 
-        schedulers = ensure_tuple(schedulers)
-        if len(schedulers) == 0:
-            schedulers = (constant_scheduler,)
-            warnings.warn(f"No scheduler was specified. Defaulting to {repr(schedulers)}")
-
-        if scale_schedule_ratio != 1.0:
-            schedulers = tuple(
-                scale_scheduler(scheduler, scale_schedule_ratio) for scheduler in ensure_tuple(schedulers))
-
         if step_schedulers_every_batch is None:
-            step_schedulers_every_batch = any(callable(scheduler) for scheduler in schedulers)
+            # only if all schedulers are ComposerSchedulers, then we can step every batch by default
+            step_schedulers_every_batch = all(
+                not isinstance(scheduler, PytorchScheduler) for scheduler in ensure_tuple(schedulers))
         self._step_schedulers_every_batch = step_schedulers_every_batch
 
-        self.state.schedulers = [compile(scheduler, self.state) for scheduler in schedulers]
+        for scheduler in ensure_tuple(schedulers):
+            if isinstance(scheduler, PytorchScheduler):
+                scale_pytorch_scheduler(scheduler, scale_schedule_ratio)
+                self.state.schedulers.append(scheduler)
+            else:  # it's a composer scheduler
+                self.state.schedulers.append(compile_composer_scheduler(scheduler, self.state, scale_schedule_ratio))
+
+        if len(self.state.schedulers) == 0:
+            warnings.warn(f"No scheduler was specified. Defaulting to {repr(schedulers)}")
```

Comment:
Let's make the error more explicit, to say that the learning rate will be constant.